### PR TITLE
Fix fill record field code action and fill record filed completions

### LINKED ISSUE #40137: [Bug]: Invalid Fill required fields code action
### Description

Consider the following scenario,

https://user-images.githubusercontent.com/61020198/230452380-bbe7c3a0-7dbf-412c-942f-062d6348ff57.mov

The `'type `field is applied again with the code action even though it already exists. 

### Steps to Reproduce

```
type Foo record {|
    int 'type;
    string name;
|};

public function testFun() {
    Foo & readonly val = {'type: 0};
}
```

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### LINKED ISSUE #40294: [Bug]: Quoted identifiers are unquoted in RecordFieldSymbol
### Description

Consider the following source.

```ballerina
type Foo record {|
    int 'type;
    string name;
|};

public function testFun() {
    Foo & readonly val = {'type: 0};
}

```

The type of variable `val` is an intersection type.The effective type of the intersection type is a `RecordType`. When investigated figured that the original name of filed `'type` is missing there. 

image

### Steps to Reproduce

_No response_

### Affected Version(s)

2201.5.0 (master)

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #40294
Fixes #40137

https://github.com/ballerina-platform/ballerina-lang/assets/46857198/f81e8d8d-ffc7-4328-8ee8-265af9f28492

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by malinthar — DISMISSED** [review]
LGTM!

File: language-server/modules/langserver-core/src/test/resources/completion/expression_context/source/mapping_constructor_expr_ctx_source8.bal [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    
+    Baz baz = {'type: "baz",};
+    
+    Foo & readonly fooReadonly = {'type: "foo",};
+    
+    record {
+        string 'type;
+        int id;
+    } & readonly fooReadonly2 = {'type: "foo",};
+}
```
Comment (SasinduDilshara) [inline language-server/modules/langserver-core/src/test/resources/completion/expression_context/source/mapping_constructor_expr_ctx_source8.bal:29]: Shall we add a one test for nested types

```
type A B;

type B type Baz record {
     readonly string 'type; 
     int id; 
 };
```

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java [hunk 2]
```diff
@@ -186,7 +186,7 @@ public static String getFillAllRecordFieldInsertText(Map<String, RecordFieldSymb
     public static Map<String, RecordFieldSymbol> getRecordFields(RawTypeSymbolWrapper<RecordTypeSymbol> wrapper,
                                                                  List<String> existingFields) {
         return wrapper.getRawType().fieldDescriptors().entrySet().stream()
-                .filter(e -> !existingFields.contains(e.getKey()))
+                .filter(e -> !existingFields.contains(e.getValue().getName().get()))
```
Comment (pcnfernando) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:189]: Can't we use the key itself anyway it's `origField.name.value`?

Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/RecordUtil.java:189]: No when we are adding the key to the map we unescaped it and add them. [link](https://github.com/ballerina-platform/ballerina-lang/blob/446f250c44fa2adfd45193986da41a459b119459/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/util/FieldMap.java#L57)

**Review by SasinduDilshara — APPROVED** [review]

**Review by malinthar — APPROVED** [review]

**Review by pcnfernando — APPROVED** [review]

**Pull request merged**